lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -59,124 +59,180 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
     state = %{file: file, column_offset: indentation + 1, braces: [], context: []}
 
     case cont do
-      :text -> handle_text(text, line, column, [], tokens, state)
-      :style -> handle_style(text, line, column, [], tokens, state)
-      :script -> handle_script(text, line, column, [], tokens, state)
+      :text -> handle_text(text, {line, -column}, column, [], tokens, state)
+      :style -> handle_style(text, {line, -column}, column, [], tokens, state)
+      :script -> handle_script(text, {line, -column}, column, [], tokens, state)
       {:comment, _, _} -> handle_comment(text, line, column, [], tokens, state)
     end
   end
 
   ## handle_text
 
-  defp handle_text("\r\n" <> rest, line, _column, buffer, acc, state) do
-    handle_text(rest, line + 1, state.column_offset, ["\r\n" | buffer], acc, state)
+  defp handle_text("\r\n" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_text(
+      rest,
+      {line + 1, offset + column - state.column_offset + 2},
+      state.column_offset,
+      ["\r\n" | buffer],
+      acc,
+      state
+    )
   end
 
-  defp handle_text("\n" <> rest, line, _column, buffer, acc, state) do
-    handle_text(rest, line + 1, state.column_offset, ["\n" | buffer], acc, state)
+  defp handle_text("\n" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_text(
+      rest,
+      {line + 1, offset + column - state.column_offset + 2},
+      state.column_offset,
+      ["\n" | buffer],
+      acc,
+      state
+    )
   end
 
-  defp handle_text("<!doctype" <> rest, line, column, buffer, acc, state) do
-    handle_doctype(rest, line, column + 9, ["<!doctype" | buffer], acc, state)
+  defp handle_text("<!doctype" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_doctype(rest, {line, offset}, column + 9, ["<!doctype" | buffer], acc, state)
   end
 
-  defp handle_text("<!DOCTYPE" <> rest, line, column, buffer, acc, state) do
-    handle_doctype(rest, line, column + 9, ["<!DOCTYPE" | buffer], acc, state)
+  defp handle_text("<!DOCTYPE" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_doctype(rest, {line, offset}, column + 9, ["<!DOCTYPE" | buffer], acc, state)
```

Comment:
No need to pattern match... we can do this throughout:

```suggestion
  defp handle_text("<!DOCTYPE" <> rest, line_offset, column, buffer, acc, state) do
    handle_doctype(rest, line_offset, column + 9, ["<!DOCTYPE" | buffer], acc, state)
```